node/core/dispute-coordinator/src/initialized.rs

```diff
@@ -861,57 +856,51 @@ impl Initialized {
 				DisputeStatus::active()
 			});
 
-			if is_confirmed {
+			if new_state.is_confirmed() {
 				*status = status.confirm();
 			}
 
 			// Note: concluded-invalid overwrites concluded-valid,
 			// so we do this check first. Dispute state machine is
 			// non-commutative.
-			if concluded_valid {
+			if new_state.is_concluded_valid() {
 				*status = status.concluded_for(now);
 			}
 
-			if concluded_invalid {
+			if new_state.is_concluded_invalid() {
 				*status = status.concluded_against(now);
 			}
 
-			Some(*status)
-		} else {
-			None
-		};
-
-		if status != prev_status {
-			if prev_status.is_none() {
-				self.metrics.on_open();
-			}
-
-			if !was_concluded_valid && concluded_valid {
-				gum::info!(
-					target: LOG_TARGET,
-					?candidate_hash,
-					session,
-					"Dispute on candidate concluded with 'valid' result",
-				);
-				self.metrics.on_concluded_valid();
-			}
-
-			if !was_concluded_invalid && concluded_invalid {
-				gum::info!(
-					target: LOG_TARGET,
-					?candidate_hash,
-					session,
-					"Dispute on candidate concluded with 'invalid' result",
-				);
-				self.metrics.on_concluded_invalid();
-			}
-
-			// Only write when updated:
 			overlay_db.write_recent_disputes(recent_disputes);
 		}
 
+		//Update metrics:
```

Comment:
```suggestion
		// Update metrics:
```

Reply:
I also suggest to extract this to a separate function that takes the `import_result`.